bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v3/ImageImpl.java

```diff
@@ -86,33 +92,60 @@ protected ImageArea newImageArea(String shape, String coordinates, String relati
         return new ImageAreaImpl(shape, coordinates, relativeCoordinates, link, alt);
     }
 
-
     @Override
     public String getSrcset() {
-        int[] widthsArray = getWidths();
-        String srcUritemplate = getSrcUriTemplate();
-        String[] srcsetArray = new String[widthsArray.length];
-        if (widthsArray.length > 0 && srcUritemplate != null) {
-            String srcUriTemplateDecoded = "";
-            try {
-                srcUriTemplateDecoded = URLDecoder.decode(srcUritemplate, StandardCharsets.UTF_8.name());
-            } catch (UnsupportedEncodingException e) {
-                LOGGER.error("Character Decoding failed for " + resource.getPath());
-            }
-            if (srcUriTemplateDecoded.contains("{.width}")) {
-                for (int i = 0; i < widthsArray.length; i++) {
-                    if (srcUriTemplateDecoded.contains("={.width}")) {
-                        srcsetArray[i] = srcUriTemplateDecoded.replace("{.width}", String.format("%s", widthsArray[i])) + " " + widthsArray[i] + "w";
-                    } else {
-                        srcsetArray[i] = srcUriTemplateDecoded.replace("{.width}", String.format(".%s", widthsArray[i])) + " " + widthsArray[i] + "w";
+        String srcUriTemplate = getSrcUriTemplate();
+        if (srcUriTemplate != null) {
+            srcUriTemplate = StringUtils.replace(srcUriTemplate, URI_WIDTH_PLACEHOLDER_ENCODED, URI_WIDTH_PLACEHOLDER);
+            if (this.isDmImage() && StringUtils.equals(this.getSmartCropRendition(), SMART_CROP_AUTO)) {
+                Map<String, String> dmSrcSetMap = getDMSrcSetMap();
+                if (!dmSrcSetMap.isEmpty()) {
+                    String[] srcsetArray = new String[dmSrcSetMap.size()];
+                    int i = 0;
+                    for (Map.Entry<String, String> item : dmSrcSetMap.entrySet()) {
+                        srcsetArray[i] =
+                                srcUriTemplate.replace(URI_WIDTH_PLACEHOLDER, String.format(":%s", item.getKey())) + " " + item.getValue() +
+                                        "w";
```

Comment:
This won't work correctly since browser might take dpr into account while selecting the src from srcset.
i.e. In case of mobile its possible to get medium or large smart crop rendition which is wrong.
To fix it correctly it needs art direction, sizes attribute